src/global-firestore-data/courses.ts

```diff
@@ -0,0 +1,83 @@
+import { doc, getDoc } from 'firebase/firestore';
+import {
+  coursesCollection,
+  availableRostersForCoursesCollection,
+  crseIdToCatalogNbrCollection,
+} from '../firebase-config';
+
+/**
+ * This function uses the subject and number of a course to retrieve the course information from 'courses' collection
+ * @param season The season that the course is under in the user's semester view
+ * @param year The year the course is under in the user's semester view
+ * @param subject The subject of the course
+ * @param number The number of the course
+ * @returns `Promise<CornellCourseRosterCourseFullDetail>`
+ */
+export const getCourseWithSeasonAndYear = async (
+  season: FirestoreSemesterSeason,
+  year: number,
+  subject: string,
+  number: string
+): Promise<CornellCourseRosterCourseFullDetail> => {
+  const roster = seasonAndYearToRosterIdentifier(season, year);
+  const course = await getDoc(doc(coursesCollection, `${roster}/${subject}/${number}`));
+  if (!course.exists()) {
+    const availableRostersForCourse = (
+      await getDoc(doc(availableRostersForCoursesCollection, `${subject} ${number}`))
+    ).data() as { rosters: string[] };
+    const lastRoster = availableRostersForCourse.rosters.length - 1;
+    const latestCourse = await getDoc(
+      doc(
+        coursesCollection,
+        `${availableRostersForCourse.rosters[lastRoster]}/${subject}/${number}`
+      )
+    );
+    return latestCourse.data()?.course as CornellCourseRosterCourseFullDetail;
+  }
+  return course.data()?.course as CornellCourseRosterCourseFullDetail;
+};
+
+/**
+ * This function uses the crseId and desired roster of a course to retrieve the course information from 'courses' collection
+ * @param roster The roster from which we want to fetch the course
+ * @param crseId  The course id of the course
+ * @returns `Promise<CornellCourseRosterCourseFullDetail>`
+ */
+export const getCourseWithCrseIdAndRoster = async (
+  roster: string,
+  crseId: number
+): Promise<CornellCourseRosterCourseFullDetail> => {
+  // use crseId to retrieve course subject and code
+  const courseSubjectAndNumber = await getDoc(doc(crseIdToCatalogNbrCollection, `${crseId}`));
+  const subject = courseSubjectAndNumber.data()?.split(' ')[0];
+  const number = courseSubjectAndNumber.data()?.split(' ')[1];
```

Comment:
Can we factor out this logic into a different function and also add error checking for the case when `courseSubjectAndNumber.data()?.split(' ')` has length < 2? We should avoid having this promise be rejected with say, an index out of bounds error. Maybe we could throw a more descriptive error instead?

Reply:
Will note this down for when it is implemented in the future.

Reply:
Sounds good. For now, can we just make sure to throw a more descriptive error in this case? Of course, we can handle its propagation later.